app/src/main/java/org/p2p/wallet/infrastructure/network/interceptor/RpcSolanaInterceptor.kt

```diff
@@ -0,0 +1,120 @@
+package org.p2p.wallet.infrastructure.network.interceptor
+
+import com.google.gson.Gson
+import okhttp3.Interceptor
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import org.json.JSONTokener
+import org.p2p.wallet.infrastructure.network.data.EmptyDataException
+import org.p2p.wallet.infrastructure.network.data.ErrorCode
+import org.p2p.wallet.infrastructure.network.data.ServerError
+import org.p2p.wallet.infrastructure.network.data.ServerException
+import timber.log.Timber
+import java.io.IOException
+
+// todo: Update validation
+private const val GSON_KEY = "method"
+private const val GSON_VALUE = "getConfirmedTransaction"
+
+open class RpcSolanaInterceptor(private val gson: Gson) : Interceptor {
+
+    private val TAG = "RpcInterceptor"
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request().newBuilder().header("Content-Type", "application/json").build()
+        val response = chain.proceed(request)
+        val responseString = response.body?.string().orEmpty()
```

Comment:
!!! copy the body, string() is one-shot method that empties response !!!

Reply:
so thats why i  moved to variable )

Reply:
but if someone or smth would need to access `response.body` again it would be empty